datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnectorTask.java

```diff
@@ -175,8 +191,21 @@ public KafkaMirrorMakerConnectorTask(KafkaBasedConnectorConfig config, Datastrea
 
   @Override
   protected void consumerSubscribe() {
-    LOG.info("About to subscribe to source: {}", _mirrorMakerSource.getTopicName());
-    _consumer.subscribe(Pattern.compile(_mirrorMakerSource.getTopicName()), this);
+    if (_enablePartitionAssignment) {
+      Set<TopicPartition> topicPartition  = getAssignedTopicPartitionFromTask();
+      updateConsumerAssignment(topicPartition);
+      // Consumer can be assigned with an empty set, but it cannot run poll against it
+      // While it's allowed to assign an empty set here, the check on poll need to be performed
+      _consumer.assign(_consumerAssignment);
```

Comment:
Looks like flushless producer uses _consumer.assignment() call to get the list of partitions assigned to the current consumer. Can you just make sure that consumer.assignment() calls works correctly if we _consumer.assign() as well?

Reply:
Yes, according to Kafka api, it works with assign().
https://kafka.apache.org/10/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html#assignment--

I've verified that it works with an empty set assignment as well.